File: src/yaml_reader_cpp/src/yaml_reader_node.cpp
#include <memory>
#include <string>
#include "rclcpp/rclcpp.hpp"

class YamlReaderNode : public rclcpp::Node
{
public:
    YamlReaderNode() : Node("yaml_reader_node")
    {
        // use_sim_time 由节点自动声明，只声明其他参数
        this->declare_parameter("frames/odom", "");
        this->declare_parameter("frames/baselink", "");
        this->declare_parameter("frames/lidar", "");
        this->declare_parameter("frames/imu", "");

        // 读取并打印
        RCLCPP_INFO(this->get_logger(), "========== YAML 参数内容 ==========");

        bool use_sim_time = this->get_parameter("use_sim_time").as_bool();
        RCLCPP_INFO(this->get_logger(), "use_sim_time : %s", use_sim_time ? "true" : "false");

        std::vector<std::string> frames = {
            "frames/odom", "frames/baselink", "frames/lidar", "frames/imu"
        };
        for (const auto & frame : frames)
        {
            std::string value = this->get_parameter(frame).as_string();
            RCLCPP_INFO(this->get_logger(), "%s : %s", frame.c_str(), value.c_str());
        }

        RCLCPP_INFO(this->get_logger(), "====================================");
    }
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<YamlReaderNode>();
    node.reset();
    rclcpp::shutdown();
    return 0;
}

File: src/yaml_reader/yaml_reader/yaml_reader_node.py
import rclpy
from rclpy.node import Node


class YamlReaderNode(Node):
    def __init__(self):
        super().__init__('yaml_reader_node')

        # 声明所有参数
        self.declare_parameter('frames/odom', '')
        self.declare_parameter('frames/baselink', '')
        self.declare_parameter('frames/lidar', '')
        self.declare_parameter('frames/imu', '')

        # 读取并打印
        self.get_logger().info('========== YAML 参数内容 ==========')

        use_sim_time = self.get_parameter('use_sim_time').value
        self.get_logger().info(f'use_sim_time : {use_sim_time}')

        frames = ['frames/odom', 'frames/baselink', 'frames/lidar', 'frames/imu']
        for frame in frames:
            value = self.get_parameter(frame).value
            self.get_logger().info(f'{frame} : {value}')

        self.get_logger().info('====================================')


def main(args=None):
    rclpy.init(args=args)
    node = YamlReaderNode()
    # 打印完即可退出，也可以改成 spin 保持运行
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()
